sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-redis/src/main/java/com/azure/messaging/eventhubs/checkpointstore/redis/JedisRedisCheckpointStore.java

```diff
@@ -47,9 +48,22 @@ public Flux<PartitionOwnership> claimOwnership(List<PartitionOwnership> requeste
      */
     @Override
     public Flux<Checkpoint> listCheckpoints(String fullyQualifiedNamespace, String eventHubName, String consumerGroup) {
-        return null;
+        String prefix = prefixBuilder(fullyQualifiedNamespace, eventHubName, consumerGroup);
+        try (Jedis jedis = jedisPool.getResource()) {
+            Set<String> members = jedis.smembers(prefix);
+            jedisPool.returnResource(jedis);
+            return Flux.fromStream(members.stream().map(json ->
+            {
+                try {
+                    return jacksonAdapter.deserialize(json, Checkpoint.class, SerializerEncoding.JSON);
+                }
+                catch (IOException e) {
+                    LOGGER.error("String could not be deserialized into Checkpoint object"); //Look into ideal way to log this error
```

Comment:
there should be an exception since there is an incorrectly stored checkpoint

Reply:
Thanks for helping me figure this out. I added the exception wherever it was needed.